bindings/r/tests/testthat/_snaps/miscellaneous.md

```diff
@@ -534,3 +534,223 @@
       )
       
 
+# motivation for not having `unmatched_delimiter` 1 (#90)
+
+    Code
+      node_children_print(node)
+    Output
+      S-Expression
+      (comment [(1, 0), (1, 77)])
+      
+      Text
+      # `if ()` is invalid R code because there isn't a `condition` between the two
+      
+      S-Expression
+      (comment [(2, 0), (2, 74)])
+      
+      Text
+      # `()`. We don't want tree-sitter to think `)` is an `unmatched_delimiter`
+      
+      S-Expression
+      (comment [(3, 0), (3, 73)])
+      
+      Text
+      # expression that it can accept as `condition`, we instead want its error
+      
+      S-Expression
+      (comment [(4, 0), (4, 45)])
+      
+      Text
+      # recovery to detect the missing `condition`.
+      
+      S-Expression
+      (comment [(6, 0), (6, 77)])
+      
+      Text
+      # TODO: In the snapshot test, the `identifier` between the `()` should really
+      
+      S-Expression
+      (comment [(7, 0), (7, 63)])
+      
+      Text
+      # show as `MISSING`, but we can't detect that right now due to:
+      
+      S-Expression
+      (comment [(8, 0), (8, 56)])
+      
+      Text
+      # https://github.com/tree-sitter/tree-sitter/issues/1043
+      
+      S-Expression
+      (binary_operator [(10, 0), (14, 1)]
+        lhs: (identifier [(10, 0), (10, 3)])
+        operator: "<-" [(10, 4), (10, 6)]
+        rhs: (function_definition [(10, 7), (14, 1)]
+          name: "function" [(10, 7), (10, 15)]
+          parameters: (parameters [(10, 15), (10, 17)]
+            "(" [(10, 15), (10, 16)]
+            ")" [(10, 16), (10, 17)]
+          )
+          body: (braced_expression [(10, 18), (14, 1)]
+            "{" [(10, 18), (10, 19)]
+            body: (if_statement [(11, 2), (13, 7)]
+              "if" [(11, 2), (11, 4)]
+              "(" [(11, 5), (11, 6)]
+              condition: (identifier [(11, 6), (11, 6)])
```

Comment:
For:

```r
      foo <- function() {
        if ()
      
        1 + 1
      }
```

This `condition` is a `MISSING` `identifier` that tree-sitter has now been able to error recover on. When the bug mentioned here is fixed we will be able to detect this and call it out as an issue.

In the meantime, this used to throw some wild error locations (https://github.com/r-lib/tree-sitter-r/issues/90) which should be better now.